mlos_bench/mlos_bench/tunables/tunable.py

```diff
@@ -9,11 +9,33 @@
 import collections
 import logging
 
-from typing import Any, List, Tuple
+from typing import List, Optional, Tuple, TypedDict, Union
 
 _LOG = logging.getLogger(__name__)
 
 
+"""A tunable parameter value type alias."""
+TunableValue = Union[int, float, str]
+
+
+class TunableDict(TypedDict, total=False):
+    """
+    A typed dict for tunable parameters.
+
+    Mostly used for mypy type checking.
+
+    These are the types expected to be received from the json config.
+    """
+
+    type: str
+    description: Optional[str]
+    default: TunableValue
+    values: Optional[List[str]]
+    # For convenience, we allow the range to be specified as a list in the json, but recast it to a tuple internally.
+    range: Optional[Union[Tuple[int, int], List[int], Tuple[float, float], List[float]]]
```

Comment:
Probably could just use `Sequence` here, but hadn't known about that yet.
Honestly, there's a bunch of our types that could get converted to less explicit types like `Mapping` instead of `Dict`.

Reply:
Did this too.